### PR TITLE
Created the Block Details page

### LINKED ISSUE #7: Block details page
# Block details page

## ⚠️ Reading contributors guidelines to get assigned is MANDATORY!

[Read contributors guidelines](https://github.com/walnuthq/op-scan/issues/1)

## User stories

As a user, when I access a block details page, I want to see basic info about the block such as block number, block hash, mining timestamp, number of transactions contained in the block as well as total gas used by the block.
When I click on previous block / next block buttons I want to navigate between adjacent blocks.
When I click on the number of transactions I want to be redirected to the block transactions list: `/block/:number/txs`.

## Validation

It should look like the Etherscan block detail page, eg. https://optimistic.etherscan.io/block/121511346
Do NOT use an accordion to show more details, just display block hash below block number.
Do NOT implement the block transactions list page.

## Implementation

You can use the transaction details page as inspiration for the global architecture.
The URL of the page should be `/block/:number`.
You should use a description list similar to the tx details page to display the block attributes and the pagination component for the previous / next block buttons.

## Resources

- tx details page: https://github.com/walnuthq/op-scan/blob/main/src/components/pages/tx/index.tsx
- shadcn/ui pagination: https://ui.shadcn.com/docs/components/pagination

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
closes #7 

After the corrections, this is what it looks like 

image

### CODE REVIEW

File: src/app/block/[number]/txs/page.tsx
Comment (saimeunt) [inline src/app/block/[number]/txs/page.tsx:1]: This should not be removed, move your [block]/page.tsx to [number]/page.tsx instead.
**The block slug is number, not block.**

File: src/app/block/[block]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+              <Link
+                href={`/block/${blockNumber}/txs`}
+                className="text-blue-600 hover:underline"
+              >
+                {blockData.transactions.length} transactions
+              </Link>
+            </DescriptionListItem>
+            <DescriptionListItem title="Gas Used">
+              {formatEther(blockData.gasUsed).toString()} (
+              {formatPercent(parseInt(blockData.gasUsed.toString()))})
```
Comment (saimeunt) [inline src/app/block/[block]/page.tsx]: This is not what we should display, see Etherscan as a reference:

Screenshot 2024-06-29 at 22 45 30

We want to display the percentage of gas used over the gas limit.
You can get inspiration from the work done on the latest blocks page:
https://github.com/walnuthq/op-scan/blob/main/src/components/pages/blocks/latest-blocks-table.tsx#L39-L46

File: src/app/block/[block]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            </DescriptionListItem>
+            <DescriptionListItem title="Gas Limit">
+              {formatEther(blockData.gasLimit).toString()}
```
Comment (saimeunt) [inline src/app/block/[block]/page.tsx]: Again, this is not what we want, stick to the exact same display as Etherscan.
See this as an inspiration:
https://github.com/walnuthq/op-scan/blob/main/src/components/pages/blocks/latest-blocks-table.tsx#L108

**Review by saimeunt — CHANGES_REQUESTED** [review]
@Dprof-in-tech Still not looking the same as Etherscan, please make the appropriate changes.

**Comment by Dprof-in-tech** [comment]
Okay chief
I’m working on it

File: src/app/block/[number]/txs/page.tsx
Comment (Dprof-in-tech) [reply src/app/block/[number]/txs/page.tsx:1]: done

File: src/app/block/[block]/page.tsx (hunk 2 above)
Comment (Dprof-in-tech) [reply src/app/block/[block]/page.tsx]: fixed

File: src/app/block/[block]/page.tsx (hunk 1 above)
Comment (Dprof-in-tech) [reply src/app/block/[block]/page.tsx]: fixed

**Comment by Dprof-in-tech** [comment]
it currently looks like this, just like the etherscan example

image

**Review by saimeunt — APPROVED** [review]

**Pull request merged**